### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by cgrosman** [devoteamgcloud/terraform-provider-looker#29]
When using the plugin (version 0.2.2) to read multiple users' data, it panics and crashes.

**Resources Configuration:**

```
data "looker_user" "iter_user" {
  for_each = var.user_ids
  id = each.key
}
```

where the `user_ids` variable is a set of strings.

**Output:**

```
module.looker-terraform-test.data.looker_user.iter_user["5441"]: Reading...
module.looker-terraform-test.data.looker_user.iter_user["6842"]: Reading...
module.looker-terraform-test.data.looker_user.iter_user["6335"]: Reading...
Error: Plugin did not respond
  with module.looker-terraform-test.data.looker_user.iter_user["6335"],
  on ../looker-role/main.tf line 10, in data "looker_user" "iter_user":
  10: data "looker_user" "iter_user" {
The plugin encountered an error, and failed to respond to the
plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more
details.
Error: Plugin did not respond
  with module.looker-terraform-test.data.looker_user.iter_user["6842"],
  on ../looker-role/main.tf line 10, in data "looker_user" "iter_user":
  10: data "looker_user" "iter_user" {
The plugin encountered an error, and failed to respond to the
plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more
details.
Error: Plugin did not respond
  with module.looker-terraform-test.data.looker_user.iter_user["5441"],
  on ../looker-role/main.tf line 10, in data "looker_user" "iter_user":
  10: data "looker_user" "iter_user" {
The plugin encountered an error, and failed to respond to the
plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more
details.
Stack trace from the terraform-provider-looker_v0.2.2 plugin:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xa98cb4]
goroutine 66 [running]:
github.com/devoteamgcloud/terraform-provider-looker/internal/provider.dataSourceUserRead({0xeb6eb8, 0xc0000d1b60}, 0x7efe8e7765b8?, {0xc0b2c0?, 0xc0004e9950?})
	github.com/devoteamgcloud/terraform-provider-looker/internal/provider/dataSource_user.go:69 +0x234
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc0004261c0, {0xeb6ef0, 0xc000090660}, 0xd?, {0xc0b2c0, 0xc0004e9950})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.17.0/helper/schema/resource.go:724 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc0004261c0, {0xeb6ef0, 0xc000090660}, 0xc000289900, {0xc0b2c0, 0xc0004e9950})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.17.0/helper/schema/resource.go:943 +0x[14](https://github.com/YotpoLtd/terraform-looker/actions/runs/4122000278/jobs/7118261312#step:6:15)5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc00000d1d0, {0xeb6e48?, 0xc000092940?}, 0xc0001492a0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.[17](https://github.com/YotpoLtd/terraform-looker/actions/runs/4122000278/jobs/7118261312#step:6:18).0/helper/schema/grpc_provider.go:1179 +0x38f
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0xc0000ca5a0, {0xeb6ef0?, 0xc0001a7f50?}, 0xc000140a00)
	github.com/hashicorp/terraform-plugin-go@v0.9.1/tfprotov5/tf5server/server.go:657 +0x409
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0xd3c240?, 0xc0000ca5a0}, {0xeb6ef0, 0xc0001a7f50}, 0xc0000d1740, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.9.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:421 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000aa700, {0xeba6c8, 0xc00029d6c0}, 0xc0002a59e0, 0xc000440db0, 0x1491a30, 0x0)
	google.golang.org/grpc@v1.47.0/server.go:1283 +0xcfd
google.golang.org/grpc.(*Server).handleStream(0xc0000aa700, {0xeba6c8, 0xc00029d6c0}, 0xc0002a59e0, 0x0)
	google.golang.org/grpc@v1.47.0/server.go:16[20](https://github.com/YotpoLtd/terraform-looker/actions/runs/4122000278/jobs/7118261312#step:6:21) +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.47.0/server.go:9[22](https://github.com/YotpoLtd/terraform-looker/actions/runs/4122000278/jobs/7118261312#step:6:23) +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.47.0/server.go:920 +0x[28](https://github.com/YotpoLtd/terraform-looker/actions/runs/4122000278/jobs/7118261312#step:6:29)a
Error: The terraform-provider-looker_v0.2.2 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**Comment by Nabil-Lahssini** [comment]
Hi, thank you for reporting this bug !
I just released a fix for this (v0.2.3), could you test it with your configuration and tell me if it works fine now?

**Comment by cgrosman** [comment]
Hi. Thank you for the quick fix. 
Looks like it works now as expected.
